src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ApplePkcs12Reader.cs

```diff
@@ -84,15 +84,32 @@ protected override AsymmetricAlgorithm LoadKey(ReadOnlyMemory<byte> pkcs8)
 
             if (key is RSAImplementation.RSASecurityTransforms rsa)
             {
-                return rsa.GetKeys().PrivateKey;
+                // Convert data key to legacy CSSM key that can be imported into keychain
+                byte[] rsaPrivateKey = rsa.ExportRSAPrivateKey();
+                using (PinAndClear.Track(rsaPrivateKey))
+                {
+                    return Interop.AppleCrypto.ImportEphemeralKey(rsaPrivateKey, true);
+                }
```

Comment:
Side-note for reviewers: The `GetPrivateKey` method is called from one place which disposes the returned SafeHandle. In case we return a new handle here it will dispose it and then dispose the original RSA key. In the case of DSA below we reuse the internal handle and it gets disposed twice which is harmless.